superset/db_engine_specs/impala.py

```diff
@@ -63,3 +74,117 @@ def get_schema_names(cls, inspector: Inspector) -> List[str]:
             if not row[0].startswith("_")
         ]
         return schemas
+
+    @classmethod
+    def has_implicit_cancel(cls) -> bool:
+        """
+        Return True if the live cursor handles the implicit cancelation of the query,
+        False otherise.
+
+        :return: Whether the live cursor implicitly cancels the query
+        :see: handle_cursor
+        """
+
+        return True
+
+    @classmethod
+    def execute(  # pylint: disable=unused-argument
+        cls,
+        cursor: Any,
+        query: str,
+        **kwargs: Any,
+    ) -> None:  # pylint: disable=arguments-differ
+        # kwargs = {"async": async_}
+        try:
+            cursor.execute_async(query)
```

Comment:
is the intention here to run `execute_async` if `_async=True` is passed in? If so, it looks like we should still keep the option to run this query synchronously.

Reply:
1. since the impala engine does not support async with execute, it is possible that my comments are ambiguous
2. because the impala engine is asynchronous, the query status can be obtained
<img width="703" alt="image" src="https://user-images.githubusercontent.com/10183241/208220292-f0717c3c-9e03-470e-90e0-3cdea9dbe1c1.png">

Reply:
Can you run execute if synchronous and execute_async only if _async is True? That way we can still run synchronously as well.

Reply:
Because you can't get the status of the query synchronously, you can't really stop the query, and the code is executed asynchronously by default ![image](https://user-images.githubusercontent.com/10183241/208811819-78d7d673-9b49-4b54-8b5c-88f8bff0b06e.png)